subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyModifier.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.artifacts.dsl;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.NonExtensible;
+import org.gradle.api.artifacts.ExternalModuleDependency;
+import org.gradle.api.artifacts.MinimalExternalModuleDependency;
+import org.gradle.api.artifacts.ModuleDependency;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderConvertible;
+
+import javax.inject.Inject;
+
+/**
+ * Universal APIs that are available for all {@code dependencies} blocks.
+ *
+ * This API is <strong>incubating</strong> and is likely to change until it's made stable.
+ * These methods are not intended to be implemented by end users or plugin authors.
+ *
+ * @since 8.0
+ */
+@Incubating
+@NonExtensible
+public interface DependencyModifier {
+    /**
+     * A dependency factory is used to convert supported dependency notations into {@link org.gradle.api.artifacts.Dependency} instances.
+     *
+     * @return a dependency factory
+     * @see DependencyFactory
+     * Do not implement this method.
```

Comment:
This is an important piece of info that is the least obvious thing here.  I think a `@NotUserImplementable` annotation, even if it did nothing and existed 100% as a marker, would be a much better option for this.

Reply:
My intention is to revisit this after we've reached an agreement on the new tags or how we signal this information.